src/std/test.ss

```diff
@@ -194,23 +196,43 @@
            (display-exception exn (current-error-port))))))
 
   (let (tests (!test-suite-tests suite))
-    (if (ormap (? (or !test-case-fail !test-case-error))
+    (cond
+     ((!test-suite-error suite)
+      => (lambda (exn)
+           (eprintf "~a: FAILED~n" (!test-suite-desc suite))
+           (eprintf "~a: ERROR " (!test-suite-desc suite))
+           (display-exception exn (current-error-port))))
+     ((ormap (? (or !test-case-fail !test-case-error))
                tests)
-      (begin
-        (eprintf "~a: FAILED~n" (!test-suite-desc suite))
-        (for-each print-failed tests))
-    (eprintf "~a: OK~n" (!test-suite-desc suite)))))
+      (eprintf "~a: FAILED~n" (!test-suite-desc suite))
+      (for-each print-failed tests))
+     (else
+      (eprintf "~a: OK~n" (!test-suite-desc suite))))))
 
 (def (test-begin!)
   (set! *tests* []))
 
 (def (test-add-test! suite)
   (push! suite *tests*))
 
+(def (with-stack-trace thunk)
```

Comment:
Shouldn't that be called `call-with-stack-trace` instead? From a CL world, the function is `call-with-` and the macro is `with-`

Reply:
ehhh, it is an internal procedure, doesn't really matter!